tracetools_trace/tracetools_trace/trace.py

```diff
@@ -388,5 +393,24 @@ def work() -> int:
     return _do_work_and_report_error(work, args.session_name, do_cleanup=False)
 
 
+def record_snapshot(args: argparse.Namespace) -> int:
+    """
+    Record snapshot of a tracing session (created in snapshot mode).
+
+    On failure, the tracing session might still exist.
+
+    :param args: the arguments parsed using
+        `tracetools_trace.tools.args.add_arguments_noninteractive_control`
+    :return: the return code (0 if successful, 1 otherwise)
+    """
+    def work() -> int:
+        session_name = args.session_name
+        if args.dual_session:
+            session_name += path.RUNTIME_SESSION_SUFFIX
+        lttng.lttng_record_snapshot(session_name=session_name)
```

Comment:
If a user does `ros2 trace record_snapshot session-name --dual-session`, they're intending to take a snapshot of the snapshot (not runtime) session, right? So this should be:

```suggestion
        session_name = args.session_name
        if args.dual_session:
            session_name += path.SNAPSHOT_SESSION_SUFFIX
        lttng.lttng_record_snapshot(session_name=session_name)
```

Reply:
Yeah, I made a mistake. I'll fix this!